mysite/lti/models.py

```diff
@@ -0,0 +1,94 @@
+
+    def create_links(self):
+        """Create all needed links to Django and/or UserSocialAuth"""
+        extra_data = json.loads(self.extra_data)
+        username = extra_data.get('lis_person_name_full', self.user_id)
+        first_name = extra_data.get('lis_person_name_given', '')
+        last_name = extra_data.get('lis_person_name_family', '')
+        email = extra_data.get('lis_person_contact_email_primary', '').lower()
+
+        if email:
+            social = UserSocialAuth.objects.filter(provider='email',
+                                                   uid=email)
+            if social:
+                django_user = social[0].user
+            else:
+                django_user = User.objects.filter(email=email)
+                if django_user:
+                    django_user = django_user[0]
+                else:
+                    django_user = User.objects.get_or_create(username=username,
+                                                             defaults={
+                                                                 'first_name': first_name,
+                                                                 'last_name': last_name,
+                                                                 'email': email
+                                                             })[0]
+                social = UserSocialAuth(user=django_user,
+                                        provider='email',
+                                        uid=email,
+                                        extra_data=extra_data)
+                social.save()
+        else:
+            django_user = User.objects.get_or_create(username=username,
+                                                     defaults={
+                                                         'first_name': first_name,
+                                                         'last_name': last_name,
+                                                     })[0]
+        self.django_user = django_user
+        self.save()
+
+    def login(self, request):
+        if self.django_user:
+            self.django_user.backend = 'django.contrib.auth.backends.ModelBackend'
+            login(request, self.django_user)
+
+    def enroll(self, roles, course_id):
+        """Create Role according to user roles from LTI POST"""
+        if not isinstance(roles, list):
+            roles = roles.split(',')
+        course = Course.objects.filter(id=course_id)
+        if course:
+            course = course[0]
+            for role in roles:
+                Role.objects.get_or_create(course=course,
+                                           user=self.django_user,
+                                           role=role)
+
+    def is_enrolled(self, roles, course_id):
+        """Check enroll status"""
+        if not isinstance(roles, list):
+            roles = roles.split(',')
+        course = Course.objects.filter(id=course_id)
```

Comment:
Please rewrite using `first` here and elsewhere
What if course is None, Role will be returned correctly?